Src/cmor_CV.c

```diff
@@ -1884,6 +1886,142 @@ int cmor_CV_setInstitution(cmor_CV_def_t * CV)
+                    CMOR_NORMAL, CV_KEY_LICENSE_ID, CV_Filename);
+
+        cmor_pop_traceback();
+        return (-1);
+    }
+
+    CV_lic_template = cmor_CV_search_child_key(CV_license, CV_KEY_LICENSE_TEMPLATE);
+    if (CV_lic_template == NULL) {
+        cmor_handle_error_variadic(
+                    "License attribute, \"%s\", could not be found in "
+                    "your Controlled Vocabulary file. (%s)",
+                    CMOR_NORMAL, CV_KEY_LICENSE_TEMPLATE, CV_Filename);
+        cmor_pop_traceback();
+        return (-1);
+    }
+
+    CV_lic = cmor_CV_search_child_key(CV_lic_ids, license_id);
+    if (CV_lic == NULL) {
+        cmor_handle_error_variadic(
+                    "The license_id, \"%s\", could not be found in "
+                    "your Controlled Vocabulary file. (%s)",
+                    CMOR_NORMAL, license_id, CV_Filename);
+        cmor_pop_traceback();
+        return (-1);
+    }
+
+    CV_lic_type = cmor_CV_search_child_key(CV_lic, CV_KEY_LICENSE_TYPE);
+    if (CV_lic_type == NULL) {
+        cmor_handle_error_variadic(
+                    "License attribute, \"%s\", could not be found in "
+                    "your Controlled Vocabulary file. (%s)",
+                    CMOR_NORMAL, CV_KEY_LICENSE_TYPE, CV_Filename);
+        cmor_pop_traceback();
+        return (-1);
+    }
+
+    CV_lic_url = cmor_CV_search_child_key(CV_lic, CV_KEY_LICENSE_URL);
+    if (CV_lic_url == NULL) {
+        cmor_handle_error_variadic(
+                    "License attribute, \"%s\", could not be found in "
+                    "your Controlled Vocabulary file. (%s)",
+                    CMOR_NORMAL, CV_KEY_LICENSE_URL, CV_Filename);
+        cmor_pop_traceback();
+        return (-1);
+    }
+
+    // Check if user provided license type or URL
+    has_lic_type = cmor_has_cur_dataset_attribute(GLOBAL_ATT_LICENSE_TYPE);
+    if (has_lic_type == 0) {
+        cmor_get_cur_dataset_attribute(GLOBAL_ATT_LICENSE_TYPE, license_type);
+
+        if (strncmp(license_type, CV_lic_type->szValue, CMOR_MAX_STRING) !=
+            0) {
+            cmor_handle_error_variadic(
+                "Your input attribute license type \"%s\" will be replaced with \n! "
+                "\"%s\" as defined in your Control Vocabulary file.\n! ",
+                CMOR_WARNING, license_type, CV_lic_type->szValue);
+        }
+    }
+    cmor_set_cur_dataset_attribute_internal(GLOBAL_ATT_LICENSE_TYPE, CV_lic_type->szValue, 1);
+
+    has_lic_url = cmor_has_cur_dataset_attribute(GLOBAL_ATT_LICENSE_URL);
+    if (has_lic_url == 0) {
+        cmor_get_cur_dataset_attribute(GLOBAL_ATT_LICENSE_URL, license_url);
+
+        if (strncmp(license_url, CV_lic_url->szValue, CMOR_MAX_STRING) !=
+            0) {
+            cmor_handle_error_variadic(
+                "Your input attribute license URL \"%s\" will be replaced with \n! "
+                "\"%s\" as defined in your Control Vocabulary file.\n! ",
```

Comment:
Ditto - it might be best to simply this error, replacing with a string variable defined up front - lots of duplication